tests/index.ts

```diff
@@ -272,3 +272,44 @@ test('should support the integrity field if present', () => {
     // We should not have made any change to the order of outputted lines (@yarnpkg/lockfile 1.0.0 had this bug)
     expect(yarn_lock).toBe(deduped);
 });
+
+test('prioritizes direct requirements if present', () => {
+    const yarn_lock = outdent`
+    a-package@*:
+      version "2.0.0"
+      resolved "http://example.com/a-package/2.0.0"
+
+    a-package@^1.0.0, a-package@^1.0.1, a-package@^1.0.2:
+      version "1.0.2"
+      resolved "http://example.com/a-package/1.0.2"
+
+    a-package@^0.1.0:
+      version "0.1.0"
+      resolved "http://example.com/a-package/0.1.0"
+
+    other-package@>=1.0.0:
```

Comment:
We should remove any mention to `other-package` in this test, as it is not related to the test intention.

Reply:
This is actually important as it verifies that packages that are not in package.json fall back to the `highest` strategy.